Tools/CISupport/ews-build/steps.py

```diff
@@ -565,6 +570,51 @@ def evaluateCommand(self, cmd):
         return rc
 
 
+class CheckOutPullRequest(steps.ShellSequence):
+    name = 'checkout-pull-request'
+    description = ['checking-out-pull-request']
+    descriptionDone = ['Checked out pull request']
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(CheckOutPullRequest, self).__init__(timeout=10 * 60, logEnviron=False, **kwargs)
+
+    def doStepIf(self, step):
+        return self.getProperty('github.number', False)
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(None)
+
+    @defer.inlineCallbacks
+    def run(self):
+        self.commands = []
+
+        remote = self.getProperty('github.head.repo.full_name', 'origin').split('/')[0]
+        project = self.getProperty('github.head.repo.full_name', self.getProperty('project'))
+        branch = self.getProperty('github.head.ref', 'main')
+
+        for command in [
+            ['/bin/sh', '-c', 'git remote add {} {}{}.git & true'.format(remote, GITHUB_URL, project)],
+            ['git', 'remote', 'set-url', remote, '{}{}.git'.format(GITHUB_URL, project)],
+            ['git', 'fetch', remote],
+            ['git', 'branch', '-f', branch, 'remotes/{}/{}'.format(remote, branch)],
+            ['git', 'checkout', branch],
+        ]:
+            self.commands.append(util.ShellArg(command=command, logname='stdio', haltOnFailure=True))
+
+        rc = yield super(CheckOutPullRequest, self).run()
+        if rc == FAILURE:
+            self.build.buildFinished(['Failed to checkout branch from PR {}'.format(self.getProperty('github.number'))], FAILURE)
+        return rc
+
+    def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'No pull request to checkout'}
+        if self.results != SUCCESS:
+            return {'step': 'Failed to checkout branch from pull request'}
```

Comment:
Might want to print PR number here as this would also go in build summary. Previous statement reads better:
`'Failed to checkout branch from PR {}'.format(self.getProperty('github.number'))`